tiledb/sm/serialization/query.cc

```diff
@@ -582,12 +578,20 @@ Status query_from_capnp(
       if ((var_size && (offset_size_left < fixedlen_size ||
                         data_size_left < varlen_size)) ||
           (!var_size && data_size_left < fixedlen_size)) {
+        if (user_buffers_overflowed != NULL) {
+          *user_buffers_overflowed = true;
```

Comment:
Shouldn't we return an OK status here? It's not really an error.

Reply:
If we returned OK, we would need to expose this `user_buffers_overflowed` publicly, otherwise a user would get an OK response even when the deserialization failed, and they wouldn't know why without string-checking the error message.

Reply:
But if the user buffers are full (but everything else worked) then it's not an error, right? So the user would in this case correctly get an OK status back, and they'd check the query status and see that it's incomplete.

Reply:
Buffers being to small isn't a failure through in this new logic right? Its an expected incomplete state, just like how if a user runs against a s3/local array with smaller buffer than the data slice they don't get any errors just incomplete query status.